File: algo-expert/typescript/src/arrays/easy/TournamentWinner.ts
export function tournamentWinner(competitions: string[][], results: number[]) {
  let score = new Map<string, number>();
  let maxScore = 0;
  let champion = '';
  for (let i in results) {
    const loser = competitions[i][results[i]];
    const winner = competitions[i][results[i] === 1 ? 0 : 1];
    const winnerScore = score.has(winner) ? score.get(winner)! : 0;
    const loserScore = score.has(loser) ? score.get(loser)! : 0;
    score.set(winner, winnerScore + 3);
    score.set(loser, loserScore);
    if (score.get(winner)! > maxScore) {
      maxScore = score.get(winner)!;
      champion = winner;
    }
  }
  return champion;
}

File: algo-expert/python/algo_expert/arrays/easy/tournament_winner.py
from collections import defaultdict


def tournament_winner(competitions: list, results: list) -> str:
    scoreboard = defaultdict(int)
    for i in range(len(results)):
        winner = 0 if results[i] == 1 else 1
        scoreboard[competitions[i][winner]] += 3
    return max(scoreboard, key=scoreboard.get)
